test/posts.test.js

```diff
@@ -126,15 +127,15 @@ describe('test /posts/:guid responses', () => {
 
   // tests
   it("pass a guid that doesn't exist", async () => {
-    const res = await request(app).get(`/posts/${missingGuid}`);
+    const res = await request(app).get(`/posts/${standardize(missingGuid, 'post')}`);
```

Comment:
Question about this.  Given the following data:

- a URL of `https://some-string.com/feed`
- hashed/encoded URL is `VM/xbiXC58KgqEqAdrzDZKR4d4UmXwuubY3/dNDlLhY=`

Our API can expose one of three things:

1. `/posts/VM/xbiXC58KgqEqAdrzDZKR4d4UmXwuubY3/dNDlLhY=`
1. `/posts/t:post:VM/xbiXC58KgqEqAdrzDZKR4d4UmXwuubY3/dNDlLhY=`
1. `/posts/https://some-string.com/feed`

I think we should automatically eliminate 2., because it once again leaks an internal implementation detail with Redis.  That leaves the raw URL string, or the hashed/encoded version.  Of the two I think that 3. is the better choice, and the more common.

As such, my proposal would be this:

- When we `addPost()` into Redis, we take the `guid` given in the post and immediately hash/encode it, then throw away the original.  We use this hash/encoded version instead for the rest of the time we deal with this post.
- We continue to use this new `guid` the same way we used the old one.
- When we get the array of posts from Redis, it will contain keys.  We need to `posts.map(key => key.replace(/^t:post:/, '')` and only return the guid vs. the namespaced-guid-key.
- When we call `Post.byGuid()` we will pass it this guid, and it will work with the storage/Redis code to namespace it before trying to look it up.
- For testing, we'll probably want to move the code to hash/encode the URI to the `Post` class, so that unit tests can call it.
- The process of hash/encoding and namespacing can thus be separated.

Does this make sense?  I'm thinking out loud and what I've written might have issues.  Let's discuss.

Reply:
I moved all the logic to `storage.js` so now tests and other components are completely blind to how redis stores keys. Now `/posts/` returns an array of unhashed, unencrypted guids.